src/frame/worker/UserCrypto.ts

```diff
@@ -43,6 +43,34 @@ function generateTransformKeyAndEncryptUserKey(userKeySet: KeyPairSet, derivedKe
     });
 }
 
+/**
+ * Requests augmentation on users current private key. The augmented private key is then encrypted and using the password derived key.
+ * @param {Uint8Array} privateKey
+ * @param {Uint8Array} augmentationFactor
+ */
+export function rotatePrivateKey(
+    passcode: string,
+    encryptedPrivateUserKey: Uint8Array,
+    derivedKeySalt: Uint8Array
+): Future<SDKError, {newEncryptedPrivateUserKey: Uint8Array; augmentationFactor: Uint8Array}> {
+    return loadRecrypt()
+        .flatMap((Recrypt) => {
+            return Future.gather2(decryptUserMasterPrivateKey(passcode, encryptedPrivateUserKey, derivedKeySalt), Recrypt.generatePasswordDerivedKey(passcode))
```

Comment:
Because the `decryptUserMasterPrivateKey` doesn't require `Recrypt` to work, we should do this work at the same time we do the `loadRecrypt` call. Take a look at the `changeUsersPasscode` function where we initially do a `gather2` like `return Future.gather2(decryptUserMasterPrivateKey(passcode, derivedKeySalt, encryptedPrivateUserKey), loadRecrypt())`. 

Also, you're passing the arguments here in the incorrect order, the `decryptUserMasterPrivateKey` method takes `passcode`, `salt`, and then `encryptedKey`.

Reply:
I also have some other improvements to make here that I had briefly mentioned yesterday but we should get these changes done first before we address those.

Reply:
fixed the two issues.